web/source/keymanweb.js

```diff
@@ -3681,21 +3722,30 @@
     } else {
       if(Pelem.tagName.toLowerCase() != 'iframe') {
         if(keymanweb.isKMWInput(Pelem)) {
-          var x = keymanweb.setupTouchElement(Pelem);
-
-          if(x) {
-            // if needed, let the timeout variable be: keymanweb.touchInputAddedTimer
-            window.setTimeout(function() {
-              keymanweb.updateInput(x);
-            }, 1);
-          }
+          keymanweb.setupTouchElement(Pelem);
```

Comment:
No, it's not so much a more monolithic function, as moving the identical tests we are doing in three different places (calls to `isKMWInput`) into a single place. The only changes to `setupTouchElement` are:
* the iframe test at the start
* the fallback to `setupNonKMWTouchElement` for e.g. numeric inputs

This means that we only need to call `setupTouchElement` elsewhere.

I don't understand the line 3683 comment?